rust/src/dcerpc/dcerpc.rs

```diff
@@ -0,0 +1,620 @@
+                self.uuid_list.push(uuidentry);
+                self.dcerpchdrudp = Some(header);
+                80
+            }
+            Err(nom::Err::Incomplete(_)) => {
+                // Insufficient data.
+                SCLogDebug!("Insufficient data while parsing DCERPC request");
+                -1
+            }
+            Err(_) => {
+                // Error, probably malformed data.
+                SCLogDebug!("An error occurred while parsing DCERPC request");
+                -1
+            }
+        }
+    }
+
+    pub fn parse_dcerpc_udp(&mut self, input: &[u8]) -> i32 {
+        // Call header parser first
+        let mut parsed = self.parse_dcerpc_udp_header(input);
+        if parsed == -1 {
+            return -1;
+        }
+
+        // Input left after successful header parsing should be 80 less
+        // than the original length
+        let mut input_left = input.len() as i32 - 80;
+        let pkt_type = self.get_hdr_pkt_type();
+        let fraglen = self.get_hdr_fraglen();
+        self.fraglenleft = fraglen;
+        self.create_new_tx(pkt_type);
+
+        // Parse rest of the body
+        while parsed >= DCERPC_UDP_HDR_LEN && parsed < fraglen as i32 && input_left > 0 {
+            let retval = self.parse_fragment_data(&input[parsed as usize..], input_left as u16);
+            if retval > 0 || retval > input_left as u16 {
+                parsed += retval as i32;
+                input_left -= retval as i32;
+            } else if input_left > 0 {
+                SCLogDebug!("Error parsing DCERPC UDP Fragment Data");
+                parsed -= input_left;
+                input_left = 0;
+            }
+        }
+        return 1;
+    }
+}
+
+pub fn evaluate_stub_params(
+    input: &[u8],
+    input_len: u16,
+    hdrflags1: u8,
+    fraglenleft: u16,
+    stub_data_buffer: &mut Vec<u8>,
+    stub_data_buffer_len: &mut u16,
+) -> u16 {
+    let stub_len: u16;
+    stub_len = cmp::min(fraglenleft, input_len);
+    if stub_len == 0 {
+        return 0;
+    }
+    // If the frag is the the first frag irrespective of it being a part of
+    // a multi frag PDU or not, it indicates the previous PDU's stub would
+    // have been buffered and processed and we can use the buffer to hold
+    // frags from a fresh request/response
+    if hdrflags1 & PFC_FIRST_FRAG > 0 {
+        *stub_data_buffer_len = 0;
+    }
+
+    let input_slice = &input[..stub_len as usize];
+    stub_data_buffer.extend_from_slice(&input_slice);
+    *stub_data_buffer_len += stub_len;
+
+    stub_len
+}
+
+#[no_mangle]
+pub extern "C" fn rs_dcerpc_udp_parse(
+    _flow: *mut core::Flow,
+    state: &mut DCERPCUDPState,
+    _pstate: *mut std::os::raw::c_void,
+    input: *const u8,
+    input_len: u32,
+    _data: *mut std::os::raw::c_void,
+    _flags: u8,
+) -> i32 {
+    if input_len > 0 {
+        if input != std::ptr::null_mut() {
+            let buf = unsafe { std::slice::from_raw_parts(input, input_len as usize) };
```

Comment:
I guess its now idiomatic to mark the whole function unsafe when dealing with raw pointers as clippy points this out:
```
12474 error: this public function dereferences a raw pointer but is not marked `unsafe`
12475    --> src/dcerpc/dcerpc.rs:381:59
12476     |
12477 381 |             let buf = unsafe { std::slice::from_raw_parts(input, input_len as usize) };
12478     |                                                           ^^^^^
12479     |
12480     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#not_unsaf      e_ptr_arg_deref
```

I think it makes sense in our code to follow this guideline of making the function unsafe, as our unsafe functions are generally short and easy to review.